# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1421860733 18000
#      Wed Jan 21 12:18:53 2015 -0500
# Node ID 228cfaf24208e350cb65ce93858698e80b83c905
# Parent  6ba5f1ff041e983337b05e78d69bfe9c5cfbc29e
put custom CPPFLAGS after AM_CPPFLAGS for dld functions (bug #44060)

* libinterp/dldfcn/config-module.awk: Put AM_CPPFLAGS ahead of
per-module CPPFLAGS.

diff --git a/libinterp/dldfcn/config-module.awk b/libinterp/dldfcn/config-module.awk
--- a/libinterp/dldfcn/config-module.awk
+++ b/libinterp/dldfcn/config-module.awk
@@ -61,17 +61,17 @@ BEGIN {
     sub (/\.cc$/, "", basename);
     print "";
     printf ("dldfcn_%s_la_SOURCES = dldfcn/%s\n",
             basename, files[i]);
     if (cppflags[i])
       {
         printf ("dldfcn/%s.df: CPPFLAGS += %s\n",
                 basename, cppflags[i]);
-        printf ("dldfcn_%s_la_CPPFLAGS = %s $(AM_CPPFLAGS)\n",
+        printf ("dldfcn_%s_la_CPPFLAGS = $(AM_CPPFLAGS) %s\n",
                 basename, cppflags[i]);
       }
     printf ("dldfcn_%s_la_LDFLAGS = -avoid-version -module $(NO_UNDEFINED_LDFLAG) %s $(OCT_LINK_OPTS)\n",
             basename, ldflags[i]);
     printf ("dldfcn_%s_la_LIBADD = $(DLD_LIBOCTINTERP_LIBADD) ../liboctave/liboctave.la %s $(OCT_LINK_DEPS)\n",
             basename, libraries[i]);
   }
 }
